flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -96,23 +104,24 @@ private void observeJmDeployment(
             if (status != null
                     && status.getAvailableReplicas() != null
                     && spec.getReplicas().intValue() == status.getReplicas()
-                    && spec.getReplicas().intValue() == status.getAvailableReplicas()
-                    && flinkService.isJobManagerPortReady(effectiveConfig)) {
-
-                // typically it takes a few seconds for the REST server to be ready
+                    && spec.getReplicas().intValue() == status.getAvailableReplicas()) {
+                if (flinkService.isJobManagerPortReady(effectiveConfig)) {
+                    // typically it takes a few seconds for the REST server to be ready
+                    LOG.info(
+                            "JobManager deployment {} in namespace {} port ready, waiting for the REST API...",
+                            flinkApp.getMetadata().getName(),
+                            flinkApp.getMetadata().getNamespace());
+                    deploymentStatus.setJobManagerDeploymentStatus(
+                            JobManagerDeploymentStatus.DEPLOYED_NOT_READY);
+                    return;
+                }
+            } else {
```

Comment:
We could also remove the else branch here (as we return early in other cases) to flatten the logic a bit to make it more readable.